src/Service/GatewayResourceService.php

```diff
@@ -68,6 +69,24 @@ public function getSchema(string $reference, string $pluginName): ?Entity
 
     }//end getSchema()
 
+    /**
+     * Get a object by identifier.
+     *
+     * @param string $id The id to look for.
+     *
+     * @return ObjectEntity|null
+     */
+    public function getObject(string $id): ?ObjectEntity
+    {
+        $objectEntity = $this->entityManager->getRepository('App:ObjectEntity')->find($id);
+        if ($objectEntity === null) {
+            $this->pluginLogger->error("No objectEntity found for $id.");
+        }
+
+        return $objectEntity;
+
+    }//end getObject()
+
```

Comment:
Your code is fine, but I'm not sure if this really is that much of an addition to this service.

Reply:
This is added so errors are always thrown and dont have to be thrown in every service manually